File: app/src/main/java/com/rikkei/tranning/chatapp/views/uis/login/LoginViewModel.java
package com.rikkei.tranning.chatapp.views.uis.login;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.rikkei.tranning.chatapp.services.models.User;
import com.rikkei.tranning.chatapp.services.models.loginUser;
import com.rikkei.tranning.chatapp.services.repositories.AllFriendRepositories;

import java.util.ArrayList;

public class LoginViewModel extends ViewModel {
    FirebaseAuth firebaseAuth= FirebaseAuth.getInstance();
    public MutableLiveData<String> EmailAddress = new MutableLiveData<>();
    public MutableLiveData<String> Password = new MutableLiveData<>();
    public MutableLiveData<Boolean> isOk=new MutableLiveData<>();
    private MutableLiveData<loginUser> userMutableLiveData;

    public MutableLiveData<loginUser> getUser() {

        if (userMutableLiveData == null) {
            userMutableLiveData = new MutableLiveData<>();
        }
        return userMutableLiveData;

    }
    public void onClick() {
        loginUser user = new loginUser(EmailAddress.getValue(), Password.getValue());
        userMutableLiveData.setValue(user);

    }
    public void loginFirebase(){
        firebaseAuth.signInWithEmailAndPassword(EmailAddress.getValue(),Password.getValue())
                .addOnSuccessListener(new OnSuccessListener<AuthResult>() {
                    @Override
                    public void onSuccess(AuthResult authResult) {
                        isOk.setValue(true);
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        isOk.setValue(false);
                    }
                });
    }

}

File: app/src/main/java/com/rikkei/tranning/chatapp/views/uis/login/LoginViewModel.kt
package com.rikkei.tranning.chatapp.views.uis.login

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.rikkei.tranning.chatapp.services.models.LoginUserModel
import com.rikkei.tranning.chatapp.services.repositories.AuthRepository

class LoginViewModel : ViewModel() {

    var loginStatus = MutableLiveData<LoginStatus>()
    var firebaseAuth = AuthRepository
    var EmailAddress = MutableLiveData<String>()
    var Password = MutableLiveData<String>()

    //    var isOk = MutableLiveData<Boolean>()
    private var userMutableLiveData: MutableLiveData<LoginUserModel>? = null
    val user: MutableLiveData<LoginUserModel>
        get() {
            if (userMutableLiveData == null) {
                userMutableLiveData = MutableLiveData()
            }
            return userMutableLiveData!!
        }

    fun onClick() {
//        loginStatus.value = LoginStatus.loading(true)
//        loginFirebase()
        val user = LoginUserModel(EmailAddress.value.toString(), Password.value.toString())
        if (!user.validateEmailPassword()) {
            loginStatus.value = LoginStatus.ErrorPassAndEmail(true)
        } else {
            loginFirebase()
        }
//        if ( user.isEmailValid)
//        userMutableLiveData!!.value = user
    }

    private fun loginFirebase() {
        loginStatus.value = LoginStatus.Loading(true)
        firebaseAuth.loginUser(
            EmailAddress.value!!,
            Password.value!!
        )
            .addOnSuccessListener {
                loginStatus.value = LoginStatus.IsOk(true)
                loginStatus.value = LoginStatus.Loading(false)
//                isOk.value = true
            }
            .addOnFailureListener {
//                isOk.value = false
                loginStatus.value = LoginStatus.Failure(it)
            }
    }

    sealed class LoginStatus {

        data class Loading(var loading: Boolean) : LoginStatus()

        data class IsOk(var isLogin: Boolean) : LoginStatus()

        data class Failure(var e: Throwable) : LoginStatus()

        data class ErrorPassAndEmail(var isError: Boolean) : LoginStatus()

    }


}